test/integration/storageversionmigrator/storageversionmigrator_test.go

```diff
@@ -0,0 +1,246 @@
+// 9. Shutdown conversion webhook
+// 10. Verify RV and Generations of CRs
+func TestStorageVersionMigrationWithCRD(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StorageVersionMigrator, true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, featuregate.Feature(clientgofeaturegate.InformerResourceVersion), true)()
+
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	crVersions := make(map[string]versions)
+
+	svmTest := svmSetup(ctx, t)
+	certCtx := svmTest.setupServerCert(t)
+
+	// create CRD with v1 serving and storage
+	crd := svmTest.createCRD(t, crdName, crdGroup, certCtx, v1CRDVersion)
+
+	// create CR
+	cr1 := svmTest.createCR(t, "cr1", "v1")
+	if ok := svmTest.isCRStoredAtVersion(t, "v1", cr1.GetName()); !ok {
+		t.Fatalf("CR not stored at version v1")
+	}
+	crVersions[cr1.GetName()] = versions{
+		generation:  cr1.GetGeneration(),
+		rv:          cr1.GetResourceVersion(),
+		isRVUpdated: true,
+	}
+
+	// add conversion webhook
+	shutdownServer := svmTest.createConversionWebhook(ctx, t, certCtx)
+
+	// add v2 for serving only
+	svmTest.updateCRD(t, crd.Name, v2CRDVersion)
+
+	// create another CR
+	cr2 := svmTest.createCR(t, "cr2", "v2")
+	if ok := svmTest.isCRStoredAtVersion(t, "v1", cr2.GetName()); !ok {
+		t.Fatalf("CR not stored at version v1")
+	}
+	crVersions[cr2.GetName()] = versions{
+		generation:  cr2.GetGeneration(),
+		rv:          cr2.GetResourceVersion(),
+		isRVUpdated: true,
+	}
+
+	// add v2 as storage version
+	svmTest.updateCRD(t, crd.Name, v2StorageCRDVersion)
+
+	// create CR with v1
+	cr3 := svmTest.createCR(t, "cr3", "v1")
+	if ok := svmTest.isCRStoredAtVersion(t, "v2", cr3.GetName()); !ok {
+		t.Fatalf("CR not stored at version v2")
+	}
+	crVersions[cr3.GetName()] = versions{
+		generation:  cr3.GetGeneration(),
+		rv:          cr3.GetResourceVersion(),
+		isRVUpdated: false,
+	}
+
+	// create CR with v2
+	cr4 := svmTest.createCR(t, "cr4", "v2")
+	if ok := svmTest.isCRStoredAtVersion(t, "v2", cr4.GetName()); !ok {
+		t.Fatalf("CR not stored at version v2")
+	}
+	crVersions[cr4.GetName()] = versions{
+		generation:  cr4.GetGeneration(),
+		rv:          cr4.GetResourceVersion(),
+		isRVUpdated: false,
+	}
+
+	// verify cr1 ans cr2 are still stored at v1
+	if ok := svmTest.isCRStoredAtVersion(t, "v1", cr1.GetName()); !ok {
+		t.Fatalf("CR not stored at version v1")
+	}
+	if ok := svmTest.isCRStoredAtVersion(t, "v1", cr2.GetName()); !ok {
+		t.Fatalf("CR not stored at version v1")
+	}
+
+	// migrate CRs from v1 to v2
+	svm, err := svmTest.createSVMResource(
+		ctx, t, "crdsvm",
+		svmv1alpha1.GroupVersionResource{
+			Group:    crd.Spec.Group,
+			Version:  "v1",
+			Resource: crd.Spec.Names.Plural,
+		})
+	if err != nil {
+		t.Fatalf("Failed to create SVM resource: %v", err)
+	}
+	if ok := svmTest.isCRDMigrated(ctx, t, svm.Name); !ok {
```

Comment:
This checks to be sure the SVM says that the writes have happened, but lacks the check for whether the instances themselves are stored in the expected version.